drivers/soundwire/bus_type.c

```diff
@@ -49,10 +55,11 @@ static int sdw_bus_match(struct device *dev, struct device_driver *ddrv)
 
 int sdw_slave_modalias(const struct sdw_slave *slave, char *buf, size_t size)
 {
-	/* modalias is sdw:m<mfg_id>p<part_id> */
+	/* modalias is sdw:m<mfg_id>p<part_id>v<version>c<class_id> */
 
-	return snprintf(buf, size, "sdw:m%04Xp%04X\n",
-			slave->id.mfg_id, slave->id.part_id);
+	return snprintf(buf, size, "sdw:m%04Xp%04Xv%02Xc%02X\n",
+			slave->id.mfg_id, slave->id.part_id,
+			slave->id.sdw_version, slave->id.class_id);
```

Comment:
@plbossart I  haven't checked all patches yet, but double-checking whether this doesn't break existing usages? One alternative would be to omit version and class-id if they are zero.
Update: hmm, I guess not as you update all the codecs in the next patch

Reply:
even without the update of codecs, when version and class ID are zero, there is a wildcard added for modalias, so we end-up selecting the previous drivers as usual.

Reply:
see e.g. for the rt1308 what happens

````
# modinfo snd-soc-rt1308-sdw 
filename:       /lib/modules/5.7.0-rc6-test+/kernel/sound/soc/codecs/snd-soc-rt1308-sdw.ko
license:        GPL v2
author:         Shuming Fan <shumingf@realtek.com>
description:    ASoC RT1308 driver SDW
alias:          sdw:m025Dp1308v02c*
depends:        snd-pcm,soundwire-bus,snd-soc-core,regmap-sdw
retpoline:      Y
intree:         Y
name:           snd_soc_rt1308_sdw
vermagic:       5.7.0-rc6-test+ SMP mod_unload 
````
so the wildcard is added for the class since it's irrelevant, but since we added the version 2 then it's added here.

The only problem I found is that there is no way to represent 4 bits with modalias, so here the version2 has to be shown as 02 to use a full byte.